File: src/test/java/com/koddy/server/common/fixture/StrategyFixture.java
package com.koddy.server.common.fixture;

import com.koddy.server.coffeechat.domain.model.Strategy;
import com.koddy.server.common.mock.fake.FakeEncryptor;

public enum StrategyFixture {
    ZOOM_LINK(Strategy.Type.ZOOM_LINK, "https://zoom-url/sjiwon"),
    GOOGLE_MEET_LINK(Strategy.Type.GOOGLE_MEET_LINK, "https://google-meet-url/sjiwon"),
    KAKAO_ID(Strategy.Type.KAKAO_ID, "sjiwon-kakao"),
    LINK_ID(Strategy.Type.LINK_ID, "sjiwon-line"),
    WECHAT_ID(Strategy.Type.WECHAT_ID, "sjiwon-wechat"),
    ;

    private final Strategy.Type type;
    private final String value;

    StrategyFixture(
            final Strategy.Type type,
            final String value
    ) {
        this.type = type;
        this.value = value;
    }

    public Strategy toDomain() {
        return Strategy.of(type, value, new FakeEncryptor());
    }

    public Strategy.Type getType() {
        return type;
    }

    public String getValue() {
        return value;
    }
}

File: src/test/kotlin/com/koddy/server/common/fixture/StrategyFixture.kt
package com.koddy.server.common.fixture

import com.koddy.server.coffeechat.domain.model.Strategy
import com.koddy.server.common.mock.fake.FakeEncryptor

enum class StrategyFixture(
    val type: Strategy.Type,
    val value: String,
) {
    ZOOM_LINK(Strategy.Type.ZOOM_LINK, "https://zoom-url/sjiwon"),
    GOOGLE_MEET_LINK(Strategy.Type.GOOGLE_MEET_LINK, "https://google-meet-url/sjiwon"),
    KAKAO_ID(Strategy.Type.KAKAO_ID, "sjiwon-kakao"),
    LINK_ID(Strategy.Type.LINK_ID, "sjiwon-line"),
    WECHAT_ID(Strategy.Type.WECHAT_ID, "sjiwon-wechat"),
    ;

    fun toDomain(): Strategy {
        return Strategy.of(
            type = type,
            value = value,
            encryptor = FakeEncryptor(),
        )
    }
}
